### DISCUSSION TITLE
Frontend is Blank when running Docker Container

### DISCUSSION THREAD

**Discussion by zohaibarsalan** [kinngh/shopify-node-express-mongodb-app#130] [Q&A] [answered]
Hello Kinngh, 

I am facing an issue while deploying. This is my first time doing it with Shopify so I might be doing something wrong. 

I have dockerized my frontend, backend and db (react, express and mongo) and ran them through docker-compose. I have a proxy that connects the backend to the frontend and nginx to run the frontend and the proxy. 

In my frontend, I am exposing port 80 and the output port is 3001. For my backend, the exposed port is 3000 and the output port is 3000 as well. Everything is working when I run the command `docker-compose up —build`.  The oAuth process gets completed and my app is installed on the users’ shopify store. 

The issue is after the app is installed. When a user (me in this case) opens the app in Shopify it’s just blank. My docker logs show that my frontend is working fine. But nothing appears on my screen, it’s just a white page. 

Something to note, I am running the docker container on my local machine and am using Cloudflare for the URL, when I put it on a server, I'll change the URLs to my domain. 

If you need any code or logs or anything else, please tell me. 

In the end, I want to deploy this on AWS. 

Thank you for the template

Regards,
Zohaib Arsalan

**Comment by kinngh** [comment]
I don't use docket so cannot provide any support.

**Reply by zohaibarsalan** [reply]
I see

Do you know of any reasons why my app page would show as white, while the oauth process gets done properly?

And how do you deploy your apps?

Thanks, 
Zohaib Arsalan

**Reply by kinngh** [reply]
1. You don't have to sign your name on each comment
2. I deploy on Heroku or a VPS from Hetzner with Caddy, depending on what I'm doing. I don't separate out my backend and frontend, and all the scripts are in the `package.json` - which usually is `npm run build; npm run start` and the Heroku is good to go. You don't have to use Docker there at all.

**Reply by zohaibarsalan** [reply, marked as answer]
Thank you for your help Kinngh. 

I have fixed the issue. My nginx.conf file did not have `mime` included and so my console was giving an error while running js and giving the mime error. 

Nonetheless thank you for your time and help.
